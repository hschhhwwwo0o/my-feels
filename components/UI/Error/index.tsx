import React, { Dispatch, FunctionComponent, SetStateAction, useEffect } from "react";

const Error: FunctionComponent<{ text: string; setError: Dispatch<SetStateAction<string>>; dependencies: any[] }> = ({
  text,
  dependencies,
  setError,
}) => {
  useEffect(() => {
    setError("");
  }, dependencies);

  return (
    <>
      <div role="alert" className="text-red-500 text-sm">
        {text}
      </div>
    </>
  );
};

export default Error;
