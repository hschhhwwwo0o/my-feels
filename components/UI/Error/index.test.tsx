import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Error from "./index";

test("Test Error component", () => {
  render(
    <Error setError={() => {}} dependencies={[]} text={"Error label"}>
      Error Text
    </Error>
  );
  const el: HTMLElement = screen.getByRole("alert");
  expect(el).toBeInTheDocument();
  expect(el).toMatchSnapshot();
});
