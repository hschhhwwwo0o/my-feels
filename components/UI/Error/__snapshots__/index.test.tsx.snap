// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Error component 1`] = `
<div
  class="text-red-500 text-sm"
  role="alert"
>
  Error label
</div>
`;
